File: 831A.py
n = int(input())
a = list(map(int,input().split()))[:n]
ch = 0
while(ch<n-1 and a[ch]<a[ch+1]):
    ch += 1
while(ch<n-1 and a[ch]==a[ch+1]):
    ch += 1
while(ch<n-1 and a[ch]>a[ch+1]):
    ch += 1
ch += 1
if(ch!=n):
    print("NO")
else:
    print("YES")

File: 831A.c
#include <stdio.h>

int main(){
    int n , i , ch = 0;
    scanf("%d",&n);
    int a[n];
    for(i=0;i<n;i++){
        scanf("%d",&a[i]);
    }
    while(ch<n-1 && (a[ch]<a[ch+1])){
        ch++;
    }
    while(ch<n-1 && a[ch]==a[ch+1]){
        ch++;
    }
    while(ch<n-1 && a[ch]>a[ch+1]){
        ch++;
    }
    
    ch++;
    if(ch!=n){
        printf("NO");
    }
    else
    {
        printf("YES");
    }
    

}